src/ScottPlot5/ScottPlot5/Plottables/Signal.cs

```diff
@@ -158,4 +158,24 @@ private void RenderHighDensity(RenderPack rp)
 
         rp.Canvas.DrawPath(path, paint);
     }
+
+    public DataPoint GetNearest(Coordinates location, RenderDetails renderInfo, float maxDistance = 15)
+    {
+        if (Data is IDataSource ds)
+            return DataSourceUtilities.GetNearest(ds, location, renderInfo, maxDistance, Axes.XAxis, Axes.YAxis);
+        else if (Data is IGetNearest gn)
+            return gn.GetNearest(location, renderInfo, maxDistance);
+        else
+            throw new NotImplementedException("Data does not implement IGetNearest");
```

Comment:
Update error text to say that IDataSource should be implemented for most accurate execution